src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -259,9 +266,22 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
             if (definition == null) {
                 throw new AbortException("No flow definition, cannot run");
             }
+
             Owner owner = new Owner(this);
-            
             FlowExecution newExecution = definition.create(owner, listener, getAllActions());
+
+            boolean loggedHintOverride = false;
+            if (getParent().isResumeBlocked()) {
+                if (newExecution instanceof BlockableResume) {
+                    ((BlockableResume) newExecution).setResumeBlocked(true);
```

Comment:
This looks unnecessary. The `FlowExecution` should be able to determine for itself whether or not resume is blocked, by inspecting the durability hint on its owner. That is why I say that you probably do not need a `setResumeBlocked` method.

Reply:
@jglick This is a separate feature -- https://issues.jenkins-ci.org/browse/JENKINS-33761 -- you can still resume a build with the lowest durability hint as long as the master shut down properly.

Reply:
The same consideration applies to both features I think, as detailed in the long-form comment.